corehq/apps/app_manager/management/commands/convert_mobile_ucr_refs.py

```diff
@@ -0,0 +1,78 @@
+import json
+import re
+import uuid
+from copy import deepcopy
+
+from django.core.management.base import BaseCommand
+
+from corehq.apps.app_manager.dbaccessors import get_app
+from corehq.apps.app_manager.exceptions import AppValidationError
+from corehq.apps.app_manager.tasks import prune_auto_generated_builds
+
+# Mobile UCR report references:
+# V1: `instance('reports')/reports/report[@id='xxxxxxx']`
+# V2: `instance('commcare-reports:xxxxxxx')`
+V1_REPORT_REFS = r"instance\('reports'\)/reports/report\[@id='([^']+)'\]"
+V2_REPORT_REFS = r"instance\('commcare-reports:\1'\)"
+
+# Mobile UCR column references:
+# V1: `column[@id = 'column_id']`
+# V2: `column_id`
+V1_COLUMN_REFS = r"column\[@id.?=.?'([^']+)'\]"
+V2_COLUMN_REFS = r'\1'
+
+
+class Command(BaseCommand):
+    help = 'Convert Mobile UCR references from V1 to V2'
+
+    def add_arguments(self, parser):
+        parser.add_argument('domain')
+        parser.add_argument('app_id')
+
+    def handle(self, domain, app_id, **options):
+        # Based on ApplicationBase.make_build()
+        v1_app = get_app(domain, app_id)
+        assert v1_app.copy_of is None
```

Comment:
Should we also assert that the domain has the `MOBILE_UCR` feature flag enabled and that the app already has its mobile restore version set to 2? If we update references in the app while the restore version is still set to 1 then this will break the app.

Reply:
334d26064b1